script.js

```diff
@@ -34,47 +35,9 @@ input.onchange = (evt) => {
         throw e;
       }
     }
-        // Transform filecontent according to your specifications
-    filecontent = filecontent.map(item => {
-      if (item.id === '_meta') {
-        return {
-          author: item.author,
-          name: item.name,
-          isOfficial: false,
-          id: item.id
-        };
-      } else {
-		  if (item.id) {
-			return { id: item.id }
-		  } else {
-			return { id: item }
-		  }     
-		}
-    });
-	
-    console.log(filecontent);
-    filecontent = filecontent.filter(e => e.id).map(e => e.id);
-    
-    for (let elt of rules) {
-      if (parseRule(elt, filecontent)) {
-        let msg = document.createElement("li");
-        
-        if (elt.note) {
-          msg.innerHTML = elt.note;
-        } else if (elt.warning) {
-          msg.style.color = "yellow";
-          msg.innerHTML = elt.warning;
-        } else if (elt.error) {
-          msg.style.color = "red";
-          msg.innerHTML = elt.error;
-        }
-        
-        output.appendChild(msg);
-      }
-    };
-  }
```

Comment:
Extracted to function `checkScript` below so the paste event can use this logic as well.